add 52592 fix control (#19009)
pingcap/docs: optimizer-fix-controls.md

## Optimizer Fix Controls reference

### [`52592`](https://github.com/pingcap/tidb/issues/52592) <span class="version-mark">New in v8.4.0</span>

- Default value: `OFF`
- Possible values: `ON`, `OFF`
- Controls whether to allow data retrieval via Get or BatchGet. If set to `ON`, the optimizer will force the selection of Coprocessor, while both Get and BatchGet will be disabled. Get and BatchGet do not support column projection, and in some scenarios, their execution cost is higher than Coprocessor requests, setting this to `ON` will result in better performance. The following are common cases where forcing Coprocessor selection is recommended:
	- Wide tables with many columns, where applications only query a smallsubset of columns.
	- Tables containing JSON columns with large JSON values, where queries either do not retrieve the JSON columns or only extract small portions of the JSON content.
	 

> 

Reviewer: The order is alphabetic.
We'd better add it at the end of the chapter.

- This variable sets the threshold for the optimizer's heuristic strategy to select access paths. If the estimated rows for an access path (such as `Index_A`) is much smaller than that of other access paths (default `1000` times), the optimizer skips the cost comparison and directly selects `Index_A`.
- `0` means to disable this heuristic strategy.

### [`52592`](https://github.com/pingcap/tidb/issues/52592) <span class="version-mark">New in v8.4.0</span>

- Default value: `OFF`
- Possible values: `ON`, `OFF`

> - Controls whether to allow data retrieval via Get or BatchGet. If set to `ON`, the optimizer will force the selection of Coprocessor, while both Get and BatchGet will be disabled. Get and BatchGet do not support column projection, and in some scenarios, their execution cost is higher than Coprocessor requests, setting this to `ON` will result in better performance. The following are common cases where forcing Coprocessor selection is recommended:
> 	- Wide tables with many columns, where applications only query a smallsubset of columns.
> 	- Tables containing JSON columns with large JSON values, where queries either do not retrieve the JSON columns or only extract small portions of the JSON content.

Reviewer 2: ```suggestion
- This variable controls whether to disable the `Point Get` and `Batch Point Get` operators for query execution. The default value `OFF` means that `Point Get` and `Batch Point Get` can be used for query execution. If set to `ON`, the optimizer disables `Point Get` and `Batch Point Get`, forcing the selection of Coprocessor for query execution.
- `Point Get` and `Batch Point Get` do not support column projection (that is, they cannot return only a subset of the columns), so in some scenarios, their execution efficiency might be lower than that of the Coprocessor, and setting this variable to `ON` can improve query performance. The following are recommended scenarios for setting this variable to `ON`:

    - Queries on wide tables with many columns, where only a few columns need to be retrieved.
    - Queries on tables with large JSON values, where the entire JSON column does not need to be retrieved, or only a small portion of the JSON column needs to be retrieved.
```

- `0` means to disable this heuristic strategy.

### [`52592`](https://github.com/pingcap/tidb/issues/52592) <span class="version-mark">New in v8.4.0</span>

- Default value: `OFF`
- Possible values: `ON`, `OFF`
- This variable controls whether to disable the `Point Get` and `Batch Point Get` operators for query execution. The default value `OFF` means that `Point Get` and `Batch Point Get` can be used for query execution. If set to `ON`, the optimizer disables `Point Get` and `Batch Point Get`, forcing the selection of Coprocessor for query execution.
- `Point Get` and `Batch Point Get` do not support column projection (that is, they cannot return only a subset of the columns), so in some scenarios, their execution efficiency might be lower than that of the Coprocessor, and setting this variable to `ON` can improve query performance. The following are recommended scenarios for setting this variable to `ON`:

    - Wide tables with many columns, where only a few columns are queried.

>     - Tables with large JSON values, where the JSON column is not queried, or only a small portion of the JSON column is queried.

Reviewer 2: ```suggestion
    - Tables with large JSON values, where the JSON column is not queried, or only a small portion of the JSON column is queried.

```

- `0` means to disable this heuristic strategy.

### [`52592`](https://github.com/pingcap/tidb/issues/52592) <span class="version-mark">New in v8.4.0</span>

- Default value: `OFF`
- Possible values: `ON`, `OFF`
- This variable controls whether to disable the `Point Get` and `Batch Point Get` operators for query execution. The default value `OFF` means that `Point Get` and `Batch Point Get` can be used for query execution. If set to `ON`, the optimizer disables `Point Get` and `Batch Point Get`, forcing the selection of Coprocessor for query execution.
- `Point Get` and `Batch Point Get` do not support column projection (that is, they cannot return only a subset of the columns), so in some scenarios, their execution efficiency might be lower than that of the Coprocessor, and setting this variable to `ON` can improve query performance. The following are recommended scenarios for setting this variable to `ON`:

    - Wide tables with many columns, where only a few columns are queried.
    - Tables with large JSON values, where the JSON column is not queried, or only a small portion of the JSON column is queried.

> 	 

Reviewer 2: ```suggestion

```
